builder/builder.go

```diff
@@ -609,3 +609,104 @@ func (b *Builder) BuildDeltaPacksPreviousVersions(prev, to uint32, printReport b
 	}
 	return nil
 }
+
+// BuildDeltaManifests between two versions of the mix.
+func (b *Builder) BuildDeltaManifests(from, to uint32) error {
+	var err error
+
+	if to == 0 {
+		to = b.MixVerUint32
+	} else {
+		if to > b.MixVerUint32 {
+			return errors.Errorf("--to version must be at most the latest mix version (%d)", b.MixVerUint32)
+		}
+	}
+	if from == to {
+		fmt.Println("the --from version matches the --to version, nothing to do")
+		return nil
+	} else if from > to {
+		return errors.Errorf("the --from version must be smaller than the --to version")
+	}
+
+	outputDir := filepath.Join(b.Config.Builder.ServerStateDir, "www")
+
+	toManifest, err := swupd.ParseManifestFile(filepath.Join(outputDir, fmt.Sprint(to), "Manifest.MoM"))
+	if err != nil {
+		return errors.Wrapf(err, "couldn't find manifest of target version")
+	}
+
+	fromManifest, err := swupd.ParseManifestFile(filepath.Join(outputDir, fmt.Sprint(from), "Manifest.MoM"))
+	if err != nil {
+		return errors.Wrapf(err, "couldn't find manifest of from version")
+	}
+
+	fmt.Printf("Using %d workers\n", b.NumDeltaWorkers)
+	_, err = swupd.CreateManifestDeltas(b.Config.Builder.ServerStateDir, fromManifest, toManifest, b.NumDeltaWorkers)
+	if err != nil {
+		log.Printf("%s\n", err)
+	}
+
+	return nil
+}
+
+// BuildDeltaManifestsPreviousVersions builds manifests to version from up to
+// prev versions. It walks the Manifest "previous" field to find those from versions.
+func (b *Builder) BuildDeltaManifestsPreviousVersions(prev, to uint32) error {
+	var err error
+
+	if to == 0 {
+		to = b.MixVerUint32
+	} else {
+		if to > b.MixVerUint32 {
+			return errors.Errorf("--to version must be at most the latest mix version (%d)", b.MixVerUint32)
+		}
+	}
+
+	outputDir := filepath.Join(b.Config.Builder.ServerStateDir, "www")
+	toManifest, err := swupd.ParseManifestFile(filepath.Join(outputDir, fmt.Sprint(to), "Manifest.MoM"))
+	if err != nil {
+		return errors.Wrapf(err, "couldn't find manifest of target version")
+	}
+
+	var previousManifests []*swupd.Manifest
+	cur := toManifest.Header.Previous
+	for i := uint32(0); i < prev; i++ {
```

Comment:
Could we have a separate function to generate the manifest instead? It seems to me that the logic in the loop plus the loop below is the same as the one in `BuildDeltaManifests`, so it would be nice if we could reuse the same code for both.

Reply:
So this is a similar pattern to the BuildDeltaPacks and BuildDeltaPacksPreviousVersions. Is that needing a similar refactor? I'm a bit unlucky on the code I stole from lol. I think these are a little awkward to combine though since one is using a specific version and the other is going back a count from the current. I would likely have just written it as one function that checks which argument is valid and then created a single or multiple element list and processed that the same way. Do you want me to go that route even though it diverges from how the pack functions are laid out?

Reply:
Well, if that is the case, I'm fine keeping it this way for now. If the others need a refactor, I'm ok with keeping the current patern for now do it all at once later on. This is also out of scope of this PR anyway.